client/app/scripts/liveblog-edit/directives.js

```diff
@@ -534,15 +542,74 @@ define([
-                controller: function() {
-                },
+                template: $templateCache.get('scripts/liveblog-edit/views/freetype-text.html'),
+                controller: ['$scope', function($scope) {
+                    $scope._id = _.uniqueId('text');
+                    if ($scope.initial !== undefined && $scope.text === '') {
+                        $scope.text = String($scope.initial);
+                    }
+                    if ($scope.number !== undefined) {
+                        $scope.$on('$destroy', $scope.$watch('text', function(value) {
+                                $scope.numberFlag = (value !== '') && (value != parseInt(value, 10));
+                                $scope.validation['number__' + $scope._id] = !$scope.numberFlag;
+                        }, true));
+                    }
+                    if ($scope.compulsory !== undefined) {
+                        $scope.$on('$destroy', $scope.$watch('[text,compulsory]', function(value) {
+                                $scope.compulsoryFlag = (value[0] === '' && value[1] === '');
+                                $scope.validation['compulsory__' + $scope._id] = !$scope.compulsoryFlag;
+                        }, true));
+                    }
+                    if ($scope.tandem !== undefined) {
+                        $scope.$on('$destroy', $scope.$watch('[text,tandem]', function(value) {
+                                $scope.tandemFlag = (value[0] === '' && value[1] !== '');
+                                $scope.validation['tandem__' + $scope._id] = !$scope.tandemFlag;
+                        }, true));
+                    }
+                    if ($scope.necessary !== undefined) {
+                        $scope.$on('$destroy', $scope.$watch('text', function(value) {
+                                $scope.necessaryFlag = (value === '');
+                                $scope.validation['necessary__' + $scope._id] = !$scope.necessaryFlag;
+                        }, true));
+                    }
+
+                }],
                 scope: {
-                    link: '='
+                    text: '=',
+                    // `compulsory` indicates a variable that is needed if the current value is empty.
+                    compulsory: '=',
+                    // `necessary` indicates is a variable needs to be non empty.
+                    necessary: '=',
+                    // `tandem` indicates a variable that is also needed.
+                    tandem: '=',
+                    validation: '=',
+                    number: '@',
+                    order: '@',
+                    initial: '@'
+                }
+            };
+        }])
+        .directive('freetypeLink', ['$templateCache', function($templateCache) {
+
+            return {
+                restrict: 'E',
+                template: $templateCache.get('scripts/liveblog-edit/views/freetype-link.html'),
```

Comment:
all files from `views/*` are compiled under `$templateCache` so why not use if it is there.